test/cli/test_reconfigure.rb

```diff
@@ -0,0 +1,46 @@
+require 'autoproj/test'
+require 'autoproj/cli/main'
+require 'autoproj/cli/reconfigure'
+require 'timeout'
+
+module Autoproj
+
+    module CLI
+        describe Reconfigure do
+            attr_reader :ws
+            before do
+                option_name = "custom-configuration-option"
+                default_value = "option-defaultvalue"
+
+                @ws = ws_create(make_tmpdir, partial_config: true)
+                # @doudou: What am I doing wrong here, 
```

Comment:
@doudou: I seem to be missing something here to correctly initiate the configuration setup, so suggestions welcome - I would like to get rid of the above partial_config workaround . ..

Reply:
Is it really worth to spawn a separate process for this? Why don't you just call `Configuration#reconfigure!` and verify that `BuildOption#ask` is called for a declared config?

Reply:
> Is it really worth to spawn a separate process for this?

It is a test, why should another process pose a problem? 

> Why don't you just call Configuration#reconfigure! and verify that BuildOption#ask is called for a declared config?

Wouldn't that be testing only the Configuration object, while here I want to cover the CLI?

Reply:
The other process does not pose a problem it just makes testing harder (or impossible). You end up with stuff like:

```ruby
it "reconfigure should run non interactively" do
    run_command 'reconfigure','--no-interactive'
end
```

Which is not really a UNIT test...